### DISCUSSION TITLE
Move community from Slack to Discord

### DISCUSSION THREAD

**Discussion by Bosch-0** [BitcoinDesign/Meta#111] [General]
This may be blasphemy at this stage but just wanted to throw the idea out there!

My main reasons for suggesting this:

- Slack does not scale with large communities without paying a significant sum of money which we do not have. We've reached out chat history limit on Slack which is resulting in old content being deleted. I think historical context is important to keep. 
- Slack lacks a lot of community engagement tools that Discord has (such as drop in chat channels - would be great for off-the-cuff design sessions). 
- Slack is much more suited for business communication, not so much public community building like Discord is designed for. 

For those who have not tried out Discord I would highly recommend doing so. Here is a small article on building a community on Discord: https://eledris.com/how-to-build-an-engaged-community-on-discord/. 

Below are some pros and cons:

**Pros**
- Community size limit isn't an issue (around 100k members is the cap so not a problem).
- No limit on messages - historical content will be kept until deleted. 
- Maybe personal preference but the UI/UX is nicer on Discord.
- Voice channels - could easily host all our calls right within Discord instead of using an external tool like Jitsi. Also would make collaboration more fluid as we could have drop in voice chat channels. An example could be say if I am doing some design work for Zeus I could hop in a chat channel and other community members could see I'm in there doing whatever it is I'm doing and jump in to chat / see what I'm up to.
- **Huge** amount of customization / bots we can utilize. A lightning network tipping bot is one cool bot we could use. 

**Cons**
- Maybe people don't use Discord as much as slack? Not that hard to transition though imo.

**Comment by Bosch-0** [comment]
### Game plan

**Goal:** move at least 90% of the community from Slack to the new [Discord](https://discord.gg/apMmBqqtq7) over the next 3 months.

1. Create a PR for bitcoin.design adding links to Discord alongside Slack.
    - Home page, footer, contribute, calender, projects
    - Add a persistent banner to the top of the whole site indicating the move
2. Write up a announcement message to share
3. Merge PR and announce the Discord move. Below should be done at the same time, mostly preferably in a favourable US time zone.
    - Share in #general
    - Share on our Twitter
    - Reach out to active community members individually to help usher users over to Discord (most notably community project / collaboration managers) 
    - Send out a special newsletter with details of the move (do this separate from the monthly newsletter, will get lost in the noise)
4. For the next 3 months active community members will need to regularly remind people in Slack to move over be it during casual conversations or in public conversations if relevant (don't be spammy)
    - Cross post between the two platforms, the channels are identical so can just copy and paste stuff.

Other ideas?

**Reply by Bosch-0** [reply]
What I mean by persistent banner image on website, something like this

![image](https://user-images.githubusercontent.com/55287964/186563662-dd373cd4-55b3-4867-8155-aad43dc45057.png)

**Comment by sbddesign** [comment]
Real quick, I am tagging #348 because that is an issue on this same topic.

I think that your steps 1 - 4 are good. I have no suggested improvements on that. The timeframe is a separate discussion, but I think that the steps in the plan are sound.

I'd like to get eyes from @pavlenex on this because of his general FOSS PM experience and eyes from @ConorOkus on this because of experience switching LDK from Slack to Discord. Do Bosch's steps 1 -4 sound good? Any other wisdom you can provide there?

**Comment by pavlenex** [comment]
ACK on the approach, except for the step 4. We can just have a bot/bridge to remind people, it shouldn't be us doing that manually, though that can be helpful. 

Pasting message I earlier posted

>  after using Discord for a while, I’m really liking it more than Slack. So a yes from me. We can always create a bot/bridge that will pass messages back and forth for a x amount of time (1-3 months). Or we can just have a bot that directs people to discord when they send a Slack message. It’s so annyoing to lose the chat history, and as Christoph said, having a focused passionate group of active people is probably more valuable anyway. We can continue marketing and mentionining we moved for the next few newsletters/meetings etc

Which pretty much aligns with the approach Bosch outlined.

And by the way having two separate communities may not necessarily be bad. For BTCPay we have telegram and mattermost, while most people are active on both, it attracts different audience. Telegram is used for chats and quick supports and mattermost is for complex issues dev need to help with, though in reality it's a mixture. 

So going to Discord doesn't have to mean shutting down Discord, we can just leave it, maybe people will take over, if it becomes inactive, or not valulable we can shut it down and continue on only Discord.

**Reply by Bosch-0** [reply]
A bot I'm going to try out https://github.com/thomasloupe/Slackord2

**Reply by moneyball** [reply]
LDK supported Slack and Discord for a period of ~6 months. It wasn't that bad but did create a bit of overhead. I also didn't notice 2 types of audiences or formats (very different than Telegram/Signal vs. Slack/Discord/Mattermost/Matrix). We then decided to direct everyone to Discord and that was surprisingly painless.

**Reply by ConorOkus** [reply]
I don't think we should use any bots at this early stage in the transition, Discord is already extremely questionable with regards to privacy and introducing bots just increases the attack surface - https://www.vauld.com/insights/nft-discord-hack-mee6-discord-bot-hack-triggers-a-domino-effect/

**Comment by GBKS** [comment]
A few additional thoughts & questions:

1. What options are available for welcoming people that join the Discord. On other services, I've seen pretty elaborate modals with all kinds of info, as well as verification mechanisms.
2. How does moderation work in Discord? What options do we have to deal with the occasional spam/scam? Are those settings configured properly at the moment (like disabling @channel in #general)? Someone already created a "sports-marketing" channel - channel creation seems like a potential issue. 
3. Not sure we need to port all channels 1:1 - how about focusing on the ones that have been most active and creating new ones only as needed?
4. We may still want to rely on Jitsi for video because it is web-based and does not require identification.
5. Initial reaction to the banner is that it feels flimsy and tacked on. Is it necessary to see this on every page in the guide, etc? Why not just put an explanation in the body of the home page?
6. Is there a bot that auto-posts GitHub-activity like we have in Slack?

**Reply by Bosch-0** [reply]
1. I made you mod on the Discord, if you go to settings -> Welcome screen I have one setup (though need to fix it, made some changes since making it). There is a bunch of settings around verification too, I have it basic for now which means they just need a verified email but you can fine tune things a lot. We can adjust this as necessary.

2. Only mods can make channels, I made that channel it was copied from Slack. There is loads of moderation options in settings. It's pretty basic atm.

3. Agree, I may have already deleted some very inactive ones 😆 

4. Yeah agree, the drop in channels are good for just casual hang outs / very quick chats imo. 

5. It'll just be looked over on the body of home page, I don't find it tacky 🤷🏿 

6. Yes you use webhooks for this, it's easy to setup. I don't have mod access to GitHub so I hadn't set this up yet.

**Reply by sbddesign** [reply]
> 2. How does moderation work in Discord?

I have found that Discord offers some very fine-tune moderation features, even in the free version. Example: you can create "roles" with specific permissions in specific channels of your workspace, and then apply those roles to users as needed.

> 4. We may still want to rely on Jitsi for video

Yes. I find Discord to be very lacking for video calls. It's really designed more for a bunch of people watching one person stream a video game, not a group conference call where multiple people want to share their desktops at different points in the call. And for some reason bandwidth and lag issues are always way more of an issue on Discord than on other platforms I've used.

> 5. Initial reaction to the banner is that it feels flimsy and tacked on.

I like that the notification bar is out of the way and doesn't intrude on the existing site. Does it need to be on every page? It would definitely get the word out! But if so, it would be nice to include a little close button to make it go away. Exact implementation is debatable, but I would just save a `closedDiscordNotification: true` to browser localStorage so it can remain closed between browser states and then move on with my life.

**Comment by Bosch-0** [comment]
We should also totally use Zebedee's tipping bot: https://zebedee.io/developers/bots

**Comment by ConorOkus** [comment]
ACK on the approach.

Number 4 might be a bit annoying and tedious given the number of channels we have in Slack but I think it's manageable if we all help out. 

Two things I'm hesitant about are:
 - Introducing any bots, they seem like a bit of wild wild west and from what I can tell some of the most popular ones are not even open source 
- I'm not sure if collaboration projects should have their own channel. For example, Fedimint is a monster of a project and to me makes more sense for them to have #design channel in their own Discord.

**Reply by Bosch-0** [reply]
Agree on Fedimint project, though collabs like bitcoin-core-app which are currently coordinated solely in our Slack it makes sense they have their own channel imo. Deleted the Fedimint channel for now. 

Yeah simply cross posting (with or without a bot) also really isn't that beneficial. it's the conversations that matter and that can only happen if the users are on the set platform.

**Reply by GBKS** [reply]
Isn't this up to the Fedimint contributors to decide?

**Reply by Bosch-0** [reply]
Well it's deleted on the Discord, not Slack. Will bring it up in the Slack channel

**Reply by ConorOkus** [reply]
I wouldn't say it's solely their decision to decide if they want to just create a channel in the Bitcoin Design Slack/Discord. It's way more of a community discussion IMO. Given the channel has thousands of people in it I think we should take more care when adding channels. 

I think having some channels tackling specific UX challenges that will occur cross-project could be an interesting way to go e.g `unified-qr`, `offers`, `async payments` etc

**Reply by Bosch-0** [reply]
Also keep in mind with Discord you can't not be in a channel

**Reply by GBKS** [reply]
Are channels in Discord fixed? I like how in Slack everyone can choose which channels they are exposed to. It allows for them to be a more flexible tool. Lots of activity around a topic? Set up a channel to make that flow more smoothly. Activity dies off and it becomes irrelevant, ditch it (ideally just from public view but still available for search/history).

I think channels should be a reflection of what people want to talk about and collaborate on, and not define the topics a small group of moderators think are important (with limits naturally, sports marketing is pretty off and a gigantic list of channels is a mess). What works practically will probably be a mix of both (some evergreens and some temporarily useful ones).

**Reply by Bosch-0** [reply]
We should let people know that they can ask us to make a channel to discuss certain topics, or we can make them if there is lots of buzz around a particular topic.

**Reply by ConorOkus** [reply]
Channels are not fixed in Discord and can be removed, but everyone is exposed to all channels in Discord so can get very noisy. You cannot really truly archive channels in Discord either, another reason we should be more selective of what channels get created. In most cases, if a channel is created it should be with the intention to be there forever. If there is activity around a topic I don't think it auto deserves its own channel necessarily. We have things like GitHub discussions or issues which could also be good for some of that.

**Reply by Bosch-0** [reply]
You can make it private and lock messages which kind of acts like archiving

**Reply by ConorOkus** [reply]
But then have to manually select who can see the private channel, so it's a bit silly lol

**Comment by ConorOkus** [comment]
https://www.commonroom.io/blog/discord-vs-slack-vs-discourse-as-your-community-channel/

**Comment by sbddesign** [comment]
I have a proposal for transitioning the Bitcoin Design Community to Discord. Inside, I lay out my argument for why this is a good decision as well as steps for accomplishing this. I'm seeing feedback on this from the Community. Thoughts, criticisms, questions?

[Discord Proposal](https://docs.google.com/document/d/1CIntJnCnPVEGs2Ui6nqSeNI67t4qUC_w3dxopxs5sKk/edit?usp=sharing)

**Reply by sbddesign** [reply]
There's a long discussion thread about this happening on Slack. Posting here for preservation.

<img width="877" alt="Screen Shot 2023-03-15 at 8 30 28 AM" src="https://user-images.githubusercontent.com/2414177/225310295-74e040be-3ce3-4185-8c06-1ebf2c5db3f1.png">

<img width="863" alt="Screen Shot 2023-03-15 at 8 30 51 AM" src="https://user-images.githubusercontent.com/2414177/225310332-d13b0f90-1ce6-4c14-93e5-1bdd023642ce.png">

<img width="859" alt="Screen Shot 2023-03-15 at 8 31 03 AM" src="https://user-images.githubusercontent.com/2414177/225310360-14a69d46-93f8-4c6e-ac5c-0437d2299b4e.png">

<img width="867" alt="Screen Shot 2023-03-15 at 8 31 22 AM" src="https://user-images.githubusercontent.com/2414177/225310390-dc554d92-9cb9-4ee8-85bf-72484dcae614.png">
